components/script/script_thread.rs

```diff
@@ -855,6 +855,72 @@ impl ScriptThread {
         });
     }
 
+    /// Check that two origins are "similar enough",
+    /// for now only used to prevent cross-origin JS url evaluation.
+    pub fn check_load_origin(source: &LoadOrigin, target: &ImmutableOrigin) -> bool {
```

Comment:
It's not obvious to me how this translates into concepts from the spec.

Reply:
Yes, that concept isn't in the spec yet, and this is more of an attempt to address the security issues in the current spec noted at https://github.com/whatwg/html/issues/2591

Reply:
Cite that issue?